spec.bs

```diff
@@ -3987,6 +3994,36 @@ in the [=user agent=]'s [=interest group set=] and whose [=map/values=] are [=k-
 This allows the browser to rerun portions of an auction without incurring the delay (and added side channels)
 from querying the server during an auction.
 
+<div algorithm>
+  To <dfn>create a k-anon restricted deep copy of the interest group</dfn> given an [=interest group=] |ig|:
+    1. Let |kAnonRestrictedIG| be a deep copy of |ig|.
+    1. If |ig|'s [=interest group/ads=] is not null:
+      1. Set |kAnonRestrictedIG|'s [=interest group/ads=] to an empty [=list=] of [=interest group ad=].
+      1. [=list/For each=] |igAd| of |ig|'s [=interest group/ads=]:
+        1. Let |adHashCode| be the result of running [=compute the key hash of ad=] given |ig| and |igAd|.
+        1. If [=query k-anonymity cache=] for |adHashCode| returns true:
+          1. If |igAd|'s [=interest group ad/selectable buyer and seller reporting IDs=] is not null:
+            1. Let |kAnonRestrictedSelectableReportingIds| be a new empty [=list=] of [=string=]s.
+            1. [=list/For each=] |selectableReportingId| in |igAd|'s 
+              [=interest group ad/selectable buyer and seller reporting IDs=]:
+              1. Let |reportingHashCode| be the result of [=computing the key hash of reporting ID=]
+                given |ig|, |igAd|, and |selectableReportingId|. 
+              1. If [=query k-anonymity cache=] for |reportingHashCode| returns true, then
+                [=list/Append=] |selectableReportingId| to |kAnonRestrictedSelectableReportingIds|.
+            1. Set |igAd|'s [=interest group ad/selectable buyer and seller reporting IDs=] to
+              |kAnonRestrictedSelectableReportingIds|.
+          1. [=list/Append=] |igAd| to |kAnonRestrictedIG|'s [=interest group/ads=].
+    1. If |ig|'s [=interest group/ad components=] is not null:
+      1. Set |kAnonRestrictedIG|'s [=interest group/ad components=] to an empty [=list=] of 
+        [=interest group ad=].
+      1. [=list/For each=] |igAdComponent| of |ig|'s [=interest group/ad components=]:
+        1. Let |adComponentHashCode| be the result of running [=compute the key hash of ad=] given |ig| and

```

Comment:
This should be `[=compute the key hash of component ad=]` (not `[=compute the key hash of ad=]`, per [the implementation](https://source.chromium.org/chromium/chromium/src/+/main:content/services/auction_worklet/bidder_worklet.cc;drc=e66b343b5554785a32ad988bfe5c3c524f5e1857;l=1519)).